common/transport/coap/src/main/java/org/thingsboard/server/transport/coap/CoapTransportResource.java

```diff
@@ -346,6 +348,16 @@ private void handleToServerRpcRequest(TbCoapClientState clientState, CoapExchang
                 new CoapNoOpCallback(exchange));
     }
 
+    /**
+     * Send an empty ACK if we are unable to send the full response within the timeout.
+     * If the full response is transmitted before the timeout this will not do anything.
+     * If this is triggered the full response will be sent in a separate CON/NON message.
+     * Essentially this allows the use of piggybacked responses.
+     */
+    private void deferAccept(CoapExchange exchange) {
+        transportContext.getScheduler().schedule(exchange::accept, piggybackTimeout, TimeUnit.MILLISECONDS);
```

Comment:
This is a valid feature and we will merge it, but please, add the check for piggybackTimeout. Something like this:
```
if (piggybackTimeout > 0){
scheduleAccept
} else {
exchange.accept();
}
```
This is required for backward compatibility and 0 performance degradation in case the user will set this setting to 0.

Reply:
Thanks for the review, added the check.